crates/storage/trie.rs

```diff
@@ -130,6 +130,28 @@ impl Trie {
             .unwrap_or(*EMPTY_TRIE_HASH))
     }
 
+    /// Obtain a merkle proof for the given path.
+    /// The proof will contain all the encoded nodes traversed until reaching the node where the path is stored (including this last node).
+    /// The proof will still be constructed even if the path is not stored in the trie, proving its absence.
+    #[allow(unused)]
+    pub fn get_proof(&self, path: &PathRLP) -> Result<Vec<Vec<u8>>, StoreError> {
```

Comment:
would it make sense to return a vec of nodes instead, so that the type is more explicit? Then the caller can convert it trivially to vec of u8

Reply:
The encoding for the nodes is standard, the structure of the node itself is not, returning the nodes would mean exposing our internal node structure which is private and not relevant to the user